librz/include/rz_util/rz_set.h

```diff
@@ -16,19 +16,33 @@ extern "C" {
 typedef HtSP RzSetS;
 
 RZ_API RZ_OWN RzSetS *rz_set_s_new(HtStrOption opt);
-RZ_API void set_s_free(RZ_NULLABLE RzSetS *set);
 RZ_API void rz_set_s_add(RZ_NONNULL RzSetS *set, const char *str);
 RZ_API bool rz_set_s_contains(RZ_NONNULL RzSetS *set, const char *str);
 RZ_API void rz_set_s_delete(RZ_NONNULL RzSetS *set, const char *str);
+RZ_API void rz_set_s_free(RZ_NULLABLE RzSetS *set);
+RZ_API ut32 rz_set_s_size(RZ_NONNULL RzSetS *set);
 RZ_API RZ_OWN RzPVector /*<char *>*/ *rz_set_s_to_vector(RZ_NONNULL RzSetS *set);
 
+/**
+ * \brief Foreach iterator over a RzSetS. The set elements can be accessed via \p it->kv.key.
+ */
+#define rz_set_s_foreach(s, it) ht_foreach (sp, s, it)
```

Comment:
Accessing elements of sets via `it->kv.key` is not user friendly. User is not supposed to know about internals of `RzSetS`.
```c
const char *elem;
rz_set_s_foreach(s, it, elem) { ... }
```
You will have to use modified `ht_foreach` body to implement this.